### PR TITLE
Add config to enable Java crash tracking

### PR DESCRIPTION
**Pull request by jbachorik**
### What does this PR do?

Enable Java crash tracking when the java agent is auto-injected

### Motivation

Have the crash-tracking working OOTB for Java applications.

### Possible Drawbacks / Trade-offs

A new file is created in `/datadog-lib/continuousprofiler/tmp/dd_crash_uploader.sh`

### Describe how to test/QA your changes

Extend the current e2e scenario to validate that the upload script is created at the expected location (`/datadog-lib/continuousprofiler/tmp/dd_crash_uploader.sh`)

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: releasenotes/notes/java-crash-tracking-b7b240529ec26798.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+features:
+  - |
+    Automatically enable crash tracking for Java applications
```
Comment (vboulineau) [inline releasenotes/notes/java-crash-tracking-b7b240529ec26798.yaml:11]: The changelog is meant for customers. Alone, this sentence does not make a lot of sense. It's more something like:
```
Auto-instrumentation admission controller now automatically activates crash tracking for Java applications
```

Also this change targets the Cluster Agent, so the `releasenote` should be created in `releasenote-dca` with `reno -d releasenotes-dca`

Comment (rtrieu) [reply releasenotes/notes/java-crash-tracking-b7b240529ec26798.yaml:11]: Agreeing with @vboulineau 's comment.

**Review by rtrieu — CHANGES_REQUESTED** [review]

File: releasenotes/notes/java-crash-tracking-b7b240529ec26798.yaml (hunk 1 above)
Comment (jbachorik) [reply releasenotes/notes/java-crash-tracking-b7b240529ec26798.yaml]: I changed the release note wording and move it to the proposed location

**Review by rtrieu — APPROVED** [review]
Looks good!

**Review by vboulineau — APPROVED** [review]

**Comment by jbachorik** [comment]
I am a bit alarmed by the serverless integration tests failing - for all runtimes!
I am quite confident that this change is affecting only Java and really don't see how this could make Go or Python fail, but here we go.

I tried asking in #serverless and #serverless-agent channels but no response so far. So, if someone here knows what is happening or whom to ask, I would be grateful.

**Pull request merged**